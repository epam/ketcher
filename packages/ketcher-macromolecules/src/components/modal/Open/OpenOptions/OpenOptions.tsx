/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/
import { Icon, IconName } from 'ketcher-react';
import styled from '@emotion/styled';
import { FileDrop } from './FileDrop';

const ICON_NAMES: Record<string, IconName> = {
  PASTE: 'open-window-paste-icon',
  FILE: 'open-window-upload-icon',
  IMAGE: 'image-frame',
} as const;

export type OpenOptionsProps = {
  selectClipboard: () => void;
  errorHandler: (err: string) => void;
  fileLoadHandler: (files: File[]) => void;
};

export const OpenOptionText = styled.p`
  font-size: 10px;
  text-transform: uppercase;
  color: ${({ theme }) => theme.ketcher.color.text.light};
  margin: 0;
  text-align: center;
  line-height: 12px;
`;

export const DisabledText = styled.p`
  font-size: 10px;
  margin: 0;
  text-align: center;
  color: ${({ theme }) => theme.ketcher.color.text.primary};
  opacity: 50%;
  line-height: 12px;
`;

const RootContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-start;
  margin-bottom: 20px;
  & > * {
    margin-right: 18px;
  }

  & :last-child {
    margin-right: 0;
  }
`;

const DropContainer = styled.div`
  width: 128px;
  height: 134px;
  box-shadow: 0 4px 12px rgba(103, 104, 132, 0.18);
  padding: 16px;
  border-radius: 12px;
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;

  & > p {
    margin-top: 6px;
    text-align: center;
  }

  svg {
    fill: ${({ theme }) => theme.ketcher.color.button.primary.active};
  }
`;

const OpenOptions = ({
  selectClipboard,
  fileLoadHandler,
  errorHandler,
}: OpenOptionsProps) => {
  return (
    <RootContainer>
      <DropContainer
        data-testid="paste-from-clipboard-button"
        onClick={selectClipboard}
      >
        <Icon name={ICON_NAMES.PASTE} />
        <OpenOptionText>Paste from clipboard</OpenOptionText>
      </DropContainer>
      <DropContainer>
        <FileDrop
          onDropAccepted={fileLoadHandler}
          onDropRejected={(e) => errorHandler(`Unable to accept file(s). ${e}`)}
          buttonLabel="Open from file"
          textLabel="or drag file here"
          iconName={ICON_NAMES.FILE}
        />
      </DropContainer>
      <DropContainer>
        <FileDrop
          accept="image/*"
          onDropAccepted={fileLoadHandler}
          onDropRejected={(e) => errorHandler(`Unable to accept file(s). ${e}`)}
          buttonLabel="Open from image"
          textLabel="or drag file here"
          iconName={ICON_NAMES.IMAGE}
          disabled
          disabledText="Image Recognition service is not available"
        />
      </DropContainer>
    </RootContainer>
  );
};

export { OpenOptions };
