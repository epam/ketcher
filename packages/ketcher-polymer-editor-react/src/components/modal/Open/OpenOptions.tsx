/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/
import { FC } from 'react'
import { Icon } from 'components/shared/icon'
import styled from '@emotion/styled'
import { FileDrop } from 'components/modal/Open/FileDrop'
import { DropButton } from 'components/modal/Open/DropButton'

const ICON_NAMES = {
  PASTE: 'capital-t',
  FILE: 'arrow-up'
}

export type OpenOptionsProps = {
  selectClipboard: () => void
  errorHandler: (err: string) => void
  fileLoadHandler: (files: File[]) => void
  isRecognizeDisabled: boolean
}

const RootContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-start;
  margin-bottom: 20px;
  & > * {
    margin-right: 18px;
  }

  & :last-child {
    margin-right: 0;
  }
`

const DropContainer = styled.div`
  width: 152px;
  height: 112px;
  border: 1px solid #aeaeae;
  border-radius: 4px;
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  box-sizing: border-box;
  position: relative;
  padding: 15px 6px;

  & > p {
    margin-top: 6px;
    text-align: center;
  }

  svg {
    fill: ${({ theme }) => theme.color.button.primary.active};
  }
`

export const OpenOptions: FC<OpenOptionsProps> = ({
  selectClipboard,
  fileLoadHandler,
  errorHandler
}) => {
  return (
    <RootContainer>
      <DropContainer>
        <DropButton
          clickHandler={selectClipboard}
          label="Paste from Clipboard"
        />
        <Icon name={ICON_NAMES.PASTE} />
      </DropContainer>
      <DropContainer>
        <FileDrop
          onDropAccepted={fileLoadHandler}
          onDropRejected={(e) => errorHandler(`Unable to accept file(s). ${e}`)}
          buttonLabel="Open from file"
          textLabel="or drag file here"
          iconName={ICON_NAMES.FILE}
        />
      </DropContainer>
    </RootContainer>
  )
}
