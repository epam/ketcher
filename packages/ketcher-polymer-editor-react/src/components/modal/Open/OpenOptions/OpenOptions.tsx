/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/
import { Icon, IconName } from 'ketcher-react';
import styled from '@emotion/styled';
import { FileDrop } from './FileDrop';
import { ActionButton } from 'components/shared/actionButton';

const ICON_NAMES: Record<string, IconName> = {
  PASTE: 'capital-t',
  FILE: 'arrow-upward',
};

export type OpenOptionsProps = {
  selectClipboard: () => void;
  errorHandler: (err: string) => void;
  fileLoadHandler: (files: File[]) => void;
};

const RootContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-start;
  margin-bottom: 20px;
  & > * {
    margin-right: 18px;
  }

  & :last-child {
    margin-right: 0;
  }
`;

const DropContainer = styled.div`
  width: 152px;
  height: 112px;
  border: 1px solid #aeaeae;
  border-radius: 4px;
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  box-sizing: border-box;
  position: relative;

  & > p {
    margin-top: 6px;
    text-align: center;
  }

  svg {
    fill: ${({ theme }) => theme.ketcher.color.button.primary.active};
  }
`;

const OpenOptions = ({
  selectClipboard,
  fileLoadHandler,
  errorHandler,
}: OpenOptionsProps) => {
  return (
    <RootContainer>
      <DropContainer>
        <ActionButton
          clickHandler={selectClipboard}
          label="Paste from Clipboard"
        />
        <Icon name={ICON_NAMES.PASTE} />
      </DropContainer>
      <DropContainer>
        <FileDrop
          onDropAccepted={fileLoadHandler}
          onDropRejected={(e) => errorHandler(`Unable to accept file(s). ${e}`)}
          buttonLabel="Open from file"
          textLabel="or drag file here"
          iconName={ICON_NAMES.FILE}
        />
      </DropContainer>
    </RootContainer>
  );
};

export { OpenOptions };
