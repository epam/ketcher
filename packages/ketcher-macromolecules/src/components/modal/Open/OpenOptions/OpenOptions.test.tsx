/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/
import { render, screen, fireEvent } from '@testing-library/react';

import { OpenOptions } from './OpenOptions';

const mockProps = {
  selectClipboard: jest.fn(),
  fileLoadHandler: jest.fn(),
  errorHandler: jest.fn(),
};

describe('OpenOptions component', () => {
  it('should render correctly', () => {
    expect(
      render(withThemeProvider(<OpenOptions {...mockProps} />)),
    ).toMatchSnapshot();
  });
  it('should render correctly with passed props', () => {
    render(withThemeProvider(<OpenOptions {...mockProps} />));

    expect(screen.getByText('Paste from clipboard')).toBeInTheDocument();
    expect(screen.getByText('Open from file')).toBeInTheDocument();
  });
  it('callback for Paste from clipboard button should be called after click', () => {
    render(withThemeProvider(<OpenOptions {...mockProps} />));
    const button = screen.getByText('Paste from clipboard');

    fireEvent.click(button);

    expect(mockProps.selectClipboard).toBeCalled();
  });
});
