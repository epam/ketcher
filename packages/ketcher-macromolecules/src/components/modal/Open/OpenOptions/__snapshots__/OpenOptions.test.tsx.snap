// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenOptions component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-9brf46"
      >
        <div
          class="css-uve29d"
          data-testid="paste-from-clipboard-button"
        >
          <svg
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M33.973 40H6.027A6.027 6.027 0 010 33.973V6.027A6.027 6.027 0 016.027 0h27.946A6.027 6.027 0 0140 6.027v27.946A6.027 6.027 0 0133.973 40z"
              fill="#EFF2F5"
            />
            <path
              d="M28.556 30.636H11.444V11.727h2.445v3.546H26.11v-3.546h2.445v18.91zM20 9.364c.324 0 .635.124.864.346.23.221.358.522.358.835 0 .314-.129.614-.358.836-.229.222-.54.346-.864.346-.324 0-.635-.124-.864-.346a1.162 1.162 0 01-.358-.835c0-.314.129-.615.358-.836.229-.222.54-.346.864-.346zm8.556 0h-5.11A3.673 3.673 0 0020 7a3.673 3.673 0 00-3.447 2.364h-5.109c-.648 0-1.27.249-1.728.692A2.325 2.325 0 009 11.727v18.91c0 .626.258 1.227.716 1.67.458.444 1.08.693 1.728.693h17.112c.648 0 1.27-.249 1.728-.692A2.325 2.325 0 0031 30.636V11.727c0-.627-.258-1.228-.716-1.671a2.487 2.487 0 00-1.728-.692z"
              fill="#167782"
            />
            <path
              clip-rule="evenodd"
              d="M15 20h10c.55 0 1-.45 1-1s-.45-1-1-1H15c-.55 0-1 .45-1 1s.45 1 1 1zm10 4H15c-.55 0-1-.45-1-1s.45-1 1-1h10c.55 0 1 .45 1 1s-.45 1-1 1zm0 4H15c-.55 0-1-.45-1-1s.45-1 1-1h10c.55 0 1 .45 1 1s-.45 1-1 1z"
              fill="#167782"
              fill-rule="evenodd"
            />
          </svg>
          <p
            class="css-9x63nq"
          >
            Paste from clipboard
          </p>
        </div>
        <div
          class="css-uve29d"
        >
          <div
            role="button"
            style="width: 100%; height: 100%; display: flex; align-items: center; flex-direction: column; justify-content: space-between;"
            tabindex="0"
          >
            <input
              autocomplete="off"
              style="display: none;"
              tabindex="-1"
              type="file"
            />
            <svg
              class="css-1opl6cd"
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M33.973 40H6.027A6.027 6.027 0 010 33.973V6.027A6.027 6.027 0 016.027 0h27.946A6.027 6.027 0 0140 6.027v27.946A6.027 6.027 0 0133.973 40z"
                fill="#EFF2F5"
              />
              <path
                d="M12.835 18.125l-2.477 9.5V15.75h20.048c0-.63-.249-1.234-.691-1.68a2.35 2.35 0 00-1.668-.695h-8.255L17.434 11h-7.076a2.35 2.35 0 00-1.667.696A2.383 2.383 0 008 13.375v14.25c0 .63.248 1.234.69 1.68a2.35 2.35 0 001.668.695h17.69a2.276 2.276 0 002.24-1.781L33 18.125H12.835zm15.212 9.5h-15.33l1.887-7.125h15.33l-1.887 7.125z"
                fill="#167782"
              />
            </svg>
            <div
              class="css-1b8y5ux"
            >
              <span>
                or drag file here
              </span>
            </div>
            <p
              class="css-9x63nq"
            >
              Open from file
            </p>
          </div>
        </div>
        <div
          class="css-uve29d"
        >
          <div
            role="button"
            style="width: 100%; height: 100%; display: flex; align-items: center; flex-direction: column; justify-content: space-between;"
          >
            <input
              accept="image/*"
              autocomplete="off"
              style="display: none;"
              tabindex="-1"
              type="file"
            />
            <svg
              class="css-1iqv54s"
              fill="none"
              height="38"
              viewBox="0 0 47 38"
              width="47"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 33.778A4.228 4.228 0 004.222 38h38c2.333 0 4.222-1.9 4.222-4.222V4.222A4.228 4.228 0 0042.222 0h-38A4.228 4.228 0 000 4.222v29.555z"
                fill="#EFF2F5"
              />
              <path
                d="M42.227 4.222h-38v29.556h38V4.222z"
                fill="#AFDADF"
              />
              <path
                d="M32.986 11.875L11.148 33.778h31.074V21.11l-9.236-9.236z"
                fill="#3498A4"
              />
              <path
                d="M21.114 14.778a7.389 7.389 0 11-14.778 0 7.389 7.389 0 0114.778 0z"
                fill="#39BBCA"
              />
              <path
                d="M14.98 17.087L4.227 27.841v5.937H31.67l-16.69-16.69z"
                fill="#167782"
              />
            </svg>
            <p
              class="css-1uh9od2"
            >
              Image Recognition service is not available
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-9brf46"
    >
      <div
        class="css-uve29d"
        data-testid="paste-from-clipboard-button"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M33.973 40H6.027A6.027 6.027 0 010 33.973V6.027A6.027 6.027 0 016.027 0h27.946A6.027 6.027 0 0140 6.027v27.946A6.027 6.027 0 0133.973 40z"
            fill="#EFF2F5"
          />
          <path
            d="M28.556 30.636H11.444V11.727h2.445v3.546H26.11v-3.546h2.445v18.91zM20 9.364c.324 0 .635.124.864.346.23.221.358.522.358.835 0 .314-.129.614-.358.836-.229.222-.54.346-.864.346-.324 0-.635-.124-.864-.346a1.162 1.162 0 01-.358-.835c0-.314.129-.615.358-.836.229-.222.54-.346.864-.346zm8.556 0h-5.11A3.673 3.673 0 0020 7a3.673 3.673 0 00-3.447 2.364h-5.109c-.648 0-1.27.249-1.728.692A2.325 2.325 0 009 11.727v18.91c0 .626.258 1.227.716 1.67.458.444 1.08.693 1.728.693h17.112c.648 0 1.27-.249 1.728-.692A2.325 2.325 0 0031 30.636V11.727c0-.627-.258-1.228-.716-1.671a2.487 2.487 0 00-1.728-.692z"
            fill="#167782"
          />
          <path
            clip-rule="evenodd"
            d="M15 20h10c.55 0 1-.45 1-1s-.45-1-1-1H15c-.55 0-1 .45-1 1s.45 1 1 1zm10 4H15c-.55 0-1-.45-1-1s.45-1 1-1h10c.55 0 1 .45 1 1s-.45 1-1 1zm0 4H15c-.55 0-1-.45-1-1s.45-1 1-1h10c.55 0 1 .45 1 1s-.45 1-1 1z"
            fill="#167782"
            fill-rule="evenodd"
          />
        </svg>
        <p
          class="css-9x63nq"
        >
          Paste from clipboard
        </p>
      </div>
      <div
        class="css-uve29d"
      >
        <div
          role="button"
          style="width: 100%; height: 100%; display: flex; align-items: center; flex-direction: column; justify-content: space-between;"
          tabindex="0"
        >
          <input
            autocomplete="off"
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <svg
            class="css-1opl6cd"
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M33.973 40H6.027A6.027 6.027 0 010 33.973V6.027A6.027 6.027 0 016.027 0h27.946A6.027 6.027 0 0140 6.027v27.946A6.027 6.027 0 0133.973 40z"
              fill="#EFF2F5"
            />
            <path
              d="M12.835 18.125l-2.477 9.5V15.75h20.048c0-.63-.249-1.234-.691-1.68a2.35 2.35 0 00-1.668-.695h-8.255L17.434 11h-7.076a2.35 2.35 0 00-1.667.696A2.383 2.383 0 008 13.375v14.25c0 .63.248 1.234.69 1.68a2.35 2.35 0 001.668.695h17.69a2.276 2.276 0 002.24-1.781L33 18.125H12.835zm15.212 9.5h-15.33l1.887-7.125h15.33l-1.887 7.125z"
              fill="#167782"
            />
          </svg>
          <div
            class="css-1b8y5ux"
          >
            <span>
              or drag file here
            </span>
          </div>
          <p
            class="css-9x63nq"
          >
            Open from file
          </p>
        </div>
      </div>
      <div
        class="css-uve29d"
      >
        <div
          role="button"
          style="width: 100%; height: 100%; display: flex; align-items: center; flex-direction: column; justify-content: space-between;"
        >
          <input
            accept="image/*"
            autocomplete="off"
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <svg
            class="css-1iqv54s"
            fill="none"
            height="38"
            viewBox="0 0 47 38"
            width="47"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 33.778A4.228 4.228 0 004.222 38h38c2.333 0 4.222-1.9 4.222-4.222V4.222A4.228 4.228 0 0042.222 0h-38A4.228 4.228 0 000 4.222v29.555z"
              fill="#EFF2F5"
            />
            <path
              d="M42.227 4.222h-38v29.556h38V4.222z"
              fill="#AFDADF"
            />
            <path
              d="M32.986 11.875L11.148 33.778h31.074V21.11l-9.236-9.236z"
              fill="#3498A4"
            />
            <path
              d="M21.114 14.778a7.389 7.389 0 11-14.778 0 7.389 7.389 0 0114.778 0z"
              fill="#39BBCA"
            />
            <path
              d="M14.98 17.087L4.227 27.841v5.937H31.67l-16.69-16.69z"
              fill="#167782"
            />
          </svg>
          <p
            class="css-1uh9od2"
          >
            Image Recognition service is not available
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
